### PR TITLE
[SW-1062] Tech enabler: Add browser extension typings

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/popup/pages/TipsClaim.vue [hunk 1]
```diff
@@ -153,7 +153,7 @@ export default defineComponent({
     onMounted(async () => {
       if (IS_EXTENSION && browser) {
         const [tab] = await browser.tabs.query({ active: true, currentWindow: true });
-        if (tab && validateTipUrl(tab.url)) {
+        if (tab && tab.url && validateTipUrl(tab.url)) {
```
Comment (pstachel) [inline src/popup/pages/TipsClaim.vue:156]:
```suggestion
        if (tab?.url && validateTipUrl(tab.url)) {
```

File: src/popup/router/index.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ const unbind = router.beforeEach(async (to, from, next) => {
     (
       !RUNNING_IN_POPUP
       && to.name === ROUTE_INDEX
-      && (await browser?.storage.local.get(lastRouteKey))[lastRouteKey]
+      && ((await browser?.storage.local.get(lastRouteKey)) as any)[lastRouteKey]
```
Comment (pstachel) [inline src/popup/router/index.ts:54]: aren't we able to define the type here?

Comment (peronczyk) [reply src/popup/router/index.ts:54]: Not that easy. The typing of the storage getter defines the return value to be `string | number | true | RegExp | ArrayBuffer | StorageObject | DataView | StorageArray | StorageMap | StorageSet | undefined` and the TS engine is complaining that it's not assignable to `NavigationGuardNextCallback`. I hate this `as any` situations, but after spending some time with this I didn't come up with anything meaningful :/

File: src/popup/pages/TipsClaim.vue (hunk 1 above)
Comment (peronczyk) [reply src/popup/pages/TipsClaim.vue:156]: Makes sense. Changed...

**Review by pstachel — APPROVED** [review]

**Review by CedrikNikita — APPROVED** [review]
Code changes look good. Didn't test it.

**Pull request merged**